File: producer-accesses/src/main/java/es/ull/domain/entity/Access.java
package es.ull.domain.entity;

import java.util.UUID;
import es.ull.domain.valueobject.EntryPoint;
import es.ull.domain.valueobject.ExitPoint;
import es.ull.domain.valueobject.IdentificationType;
import java.time.LocalDateTime;
import es.ull.utils.TruckUtils;

public class Access {

    private UUID id;
    private LocalDateTime entryDate;
    private LocalDateTime exitDate;
    private EntryPoint entryPoint;
    private ExitPoint exitPoint;
    private IdentificationType identificationType;
    private UUID truckID;
    private UUID portID;

    public UUID getId() {
        return this.id;
    }

    public void setId(UUID id) {
        this.id = id;
    }

    public EntryPoint getEntryPoint() {
        return this.entryPoint;
    }

    public void setEntryPoint(EntryPoint entryPoint) {
        this.entryPoint = entryPoint;
    }

    public ExitPoint getExitPoint() {
        return this.exitPoint;
    }

    public void setExitPoint(ExitPoint exitPoint) {
        this.exitPoint = exitPoint;
    }

    public LocalDateTime getEntryDate() {
        return this.entryDate;
    }

    public void setEntryDate(LocalDateTime entryDate) {
        this.entryDate = entryDate;
    }

    public LocalDateTime getExitDate() {
        return this.exitDate;
    }

    public void setExitDate(LocalDateTime exitDate) {
        this.exitDate = exitDate;
    }

    public IdentificationType getIdentificationType() {
        return this.identificationType;
    }

    public void setIdentificationType(IdentificationType identificationType) {
        this.identificationType = identificationType;
    }

    public UUID getTruckID() {
        return this.truckID;
    }

    public void setTruckID(UUID truckID) {
        this.truckID = truckID;
    }

    public UUID getPortID() {
        return this.portID;
    }

    public void setPortID(UUID portID) {
        this.portID = portID;
    }

    public static Access random() {
        Access access = new Access();
        access.setId(UUID.randomUUID());
        access.setEntryPoint(EntryPoint.random());
        access.setExitPoint(ExitPoint.random());
        access.setEntryDate(LocalDateTime.now());
        access.setExitDate(LocalDateTime.now());
        access.setIdentificationType(IdentificationType.random());
        access.setTruckID(UUID.randomUUID());
        access.setPortID(UUID.randomUUID());
        return access;
    }

    public String toJson() {
        return "{" +
            " idAccess='" + getId().toString() + "'" +
            ", idTruck='" + getTruckID().toString() + "'" +
            ", idPort='" + getPortID().toString() + "'" +
            ", entryDate='" + TruckUtils.localDateTimeToString(getEntryDate()) + "'" +
            ", exitDate='" + TruckUtils.localDateTimeToString(getExitDate()) + "'" +
            ", entryPoint='" + getEntryPoint().toString() + "'" +
            ", exitPoint='" + getExitPoint().toString() + "'" +
            ", identificationType='" + getIdentificationType().toString() + "'" +
            "}";
    }
}

File: front-end/src/domain/entity/access.ts
import { UllUUID } from '../../utils/ull-uuid';
import { Utils } from '../../utils/utils';
import { EntryPoint } from '../valueobject/entryPoint';
import { EntryPointVO } from '../valueobject/entryPoint';
import { ExitPoint } from '../valueobject/exitPoint';
import { ExitPointVO } from '../valueobject/exitPoint';
import { IdentificationType } from '../valueobject/identificationType';
import { IdentificationTypeVO } from '../valueobject/identificationType';
import { DateUtils } from '../../utils/dateUtils';

export class Access {
    
    private _id: UllUUID = UllUUID.random();
    private _entryDate: Date;
    private _exitDate: Date;
    private _entryPoint: EntryPoint;
    private _exitPoint: ExitPoint;
    private _identificationType: IdentificationType;
    private _truckID: UllUUID;
    private _portID: UllUUID;

    constructor(id: UllUUID = UllUUID.random(), EntryDate: Date, exitDate: Date, entryPoint: EntryPoint, exitPoint: ExitPoint, identificationType: IdentificationType, truck: UllUUID, port: UllUUID) {
        this._id = id;
        this._entryDate = EntryDate;
        this._exitDate = exitDate;
        this._entryPoint = entryPoint;
        this._exitPoint = exitPoint;
		this._identificationType = identificationType;
		this._truckID = truck;
		this._portID = port;
    }
    
    get id(): UllUUID {
        return this._id;
    }
    
    set id(id: UllUUID) {
        this._id = id;
    }
    
    set entryDate(entryDate: Date) {
        this._entryDate = entryDate;
    }

    get entryDate(): Date {
        return this._entryDate;
    }

    get exitDate(): Date {
        return this._exitDate;
    }

    set exitDate(exitDate: Date) {
        this._exitDate = exitDate;
    }

    get entryPoint(): EntryPoint {
        return this._entryPoint;
    }

    set entryPoint(entryPoint: EntryPoint) {
        this._entryPoint = entryPoint;
    }

	get exitPoint(): ExitPoint {
        return this._exitPoint;
    }

    set exitPoint(exitPoint: ExitPoint) {
        this._exitPoint = exitPoint;
    }

    get identificationType(): IdentificationType {
        return this._identificationType;
    }

    set identificationType(identificationType: IdentificationType) {
        this._identificationType = identificationType;
    }

	get truckID(): UllUUID {
        return this._truckID;
    }

    set truckID(truck: UllUUID) {
        this._truckID = truck;
    }

    get portID(): UllUUID {
        return this._portID;
    }

    set portID(port: UllUUID) {
        this._portID = port;
    }

    public static random(): Access {
        return new Access(
            UllUUID.random(),
        	Utils.randomDate(),
        	Utils.randomDate(),
        	EntryPointVO.random(),
        	ExitPointVO.random(),
			IdentificationTypeVO.random(),
			UllUUID.random(),
			UllUUID.random()
        );
    }

    public toJson(): any {
        return {
            "idAccess": this.id.toString(),
            "idTruck": this.truckID.toString(),
            "idPort": this.portID.toString(),
            "entryDate": DateUtils.toJson(this.entryDate),
            "exitDate": DateUtils.toJson(this.exitDate),
            "entryPoint": EntryPointVO.fromOrdinal(this.entryPoint),
            "exitPoint": ExitPointVO.fromOrdinal(this.exitPoint),
            "identificationType": IdentificationTypeVO.fromOrdinal(this.identificationType)
        };
    }

    public static fromJson(json: any): Access {
        const id = new UllUUID(json.idAccess);
        const truckID = (json.idTruck);
        const portID = (json.idPort);
        const entryDate = new Date(json.entryDate);
        const exitDate = new Date(json.exitDate);
        const entryPoint = EntryPoint[json.entryPoint as keyof typeof EntryPoint]
        const exitPoint = ExitPoint[json.exitPoint as keyof typeof ExitPoint]
        const identificationType = IdentificationType[json.identificationType as keyof typeof IdentificationType]
        return new Access(id, entryDate, exitDate, entryPoint, exitPoint, identificationType, truckID, portID);
    }
}